tempodb/blocklist/poller.go

```diff
@@ -223,12 +235,13 @@ func (p *Poller) pollTenantAndCreateIndex(
 
 		// polling fallback is true, log the error and continue in this method to completely poll the backend
 		level.Error(p.logger).Log("msg", "failed to pull bucket index for tenant. falling back to polling", "tenant", tenantID, "err", err)
+	} else {
+		metricTenantIndexBuilder.WithLabelValues(tenantID).Set(1)
```

Comment:
any particular reason to move this? is this to distinguish between a "true" tenant index builder and a compactor that's just falling through?

Reply:
This is to avoid setting the builder to true when the poller doesn't own the writing.  I think this is what we intend.  Or are you thinking that even if not configured to be the writer, if we fall though and end up writing, that we should metric as such?